src/ng/directive/form.js

```diff
@@ -497,18 +507,27 @@ var formDirectiveFactory = function(isNgForm) {
                 event.preventDefault();
               };
 
+              var handleKeypress = function(event) {
```

Comment:
That's too simple, see https://docs.angularjs.org/api/ng/directive/form#submitting-a-form-and-preventing-the-default-action
Maybe we should instead listen on the submit event and see if the first parent form of explicitOriginalTarget is 1) not the current submit target and 2) has the do-not-propagate attribute set